grammars/rainfall/ocaml.py

```diff
@@ -0,0 +1,467 @@
+from autoplan.grammar import Rule
+from .labels import GeneralRainfallLabels
+
+Labels = GeneralRainfallLabels
+
+class CleanFirst(Rule):
+    def render(self):
+        recursion = self.choice('recursion', {'' : 1, 'rec' : 1})
+        parameter_specs = self.choice('parameter_specs', {'list_name_only' : 1, 'list_name_and_type' : 1})
+        list_name = self.choice('list_name', {'alon': 1, 'aloi': 1, 'alod' : 1, 'alof' : 1, 'amounts' : 1})
```

Comment:
Just a note, if you use the default tokenization strategy, it will anonymize identifiers, so these sources of variation won't be seen by the model. (all variables look the same)